drivers/scheduler/k8s/k8s.go

```diff
@@ -249,6 +263,26 @@ func (k *K8s) String() string {
 	return SchedName
 }
 
+// String returns the version of the k8s cluster in the format (Maj, Min, Err)
+func ClusterVersion() (int64, int64, error) {
+	if ver, err := k8sCore.GetVersion(); err != nil {
+		return 0, 0, err
+	} else {
+		var err error
+		var maj, min int64
+
+		if maj, err = strconv.ParseInt(ver.Major, 10, 0); err != nil {
+			return 0, 0, err
+		}
+
+		if min, err = strconv.ParseInt(ver.Minor, 10, 0); err != nil {
+			return 0, 0, err
+		}
+
+		return maj, min, nil
+	}
+}
```

Comment:
I don't see the reason for having this function. I see that you are using this function to compare the versions of 2 k8s cluster versions.

You can create a function like this

```
// compareVersions compares two versions in semver format
// returns -1 if v1 is less than v2
// returns 0 if v1 is equal to v2
// returns 1 if v1 is greater than v2
func compareVersions(v1, v2 string) int {
	// Split the version strings into their components
	v1Parts := strings.Split(v1, ".")
	v2Parts := strings.Split(v2, ".")

	// Make sure the number of components is the same for both versions
	if len(v1Parts) != len(v2Parts) {
		return -2
	}

	// Loop through each component and compare them
	for i := 0; i < len(v1Parts); i++ {
		v1Part, err1 := strconv.Atoi(v1Parts[i])
		v2Part, err2 := strconv.Atoi(v2Parts[i])

		// If there is an error converting a component to an integer, return -2
		if err1 != nil || err2 != nil {
			return -2
		}

		// If the components are not equal, return -1 or 1 depending on which is greater
		if v1Part != v2Part {
			if v1Part < v2Part {
				return -1
			} else {
				return 1
			}
		}
	}

	// If all components are equal, return 0
	return 0
}
```

And in your test you can just call this function

```
result := compareVersions(v1, v2)

	if result == 0 {
		fmt.Printf("%s is equal to %s\n", v1, v2)
	} else if result < 0 {
		fmt.Printf("%s is less than %s\n", v1, v2)
	} else {
		fmt.Printf("%s is greater than %s\n", v1, v2)
	}
```

This looks cleaner and there is no need for a function which just returns major and minor version

Reply:
@mkoppal-px  if I am understanding this correctly, then I'll still need to leave the `ClusterVersion` function in `k8s` be, except I'll make it:
```
func ClusterVersion() (string, error) {
	if ver, err := k8sCore.GetVersion(); err != nil {
		return "", err
	} else {
		return ver.String(), nil
	}
}
```

and then I'll place the `compareVersions` function you suggested in `k8s.go` (or should it be in `backup_helper.go` as it is specifically a requirement (helper) of a test)

Reply:
@mkoppal-px  on second thoughts, I'll use the `semver` suggested by @lsrinivas-pure . It can be used in 3 lines.

Reply:
Makes sense. Please go ahead.

Reply:
✅